### PR TITLE
Landing Page + Modal & Form Framework 

### LINKED ISSUE #1301: Landing page redesign

### LINKED ISSUE #1302: Redesign modals

### LINKED ISSUE #1305: Save account type -> user.data

### LINKED ISSUE #1319: Actually verify the Captcha (set up backend)
Captcha is there to block UI submission at this point, but should be required by the server (at least in production)

### PR DESCRIPTION
**Pull request by maxbates**
#### Overview

- Landing Page ( #1301 )
  - Update routing
  - put content in frame so easy to update
  - save account type to user data (free vs paid)
- New Modal framework ( #1302 )
  - update register, signin, forgot password
  - others will be relatively straightforward
- New Form styling framework
  - several components for forms + styling

#### Type of change (bug fix, feature, docs, UI, etc.)

feature

#### Breaking Changes

Changes default passwords for tests - the modal requires alphanumeric, length > 8 (but not server API)

#### Requirements

- [x] Adds a test (for features + fixes)
- [ ] Docs updated (for features + fixes)

#### Other information

e2e tests updated

#### Issue

close #1301 
close #1302
close #1305 
close #1319

### CODE REVIEW

**Comment by maxbates** [comment]
It would probably make sense to update the "Forgot Password" Modal too

**Comment by cojofra** [comment]
looks like it already updated
screen shot 2017-01-18 at 5 51 16 pm
screen shot 2017-01-18 at 5 50 20 pm

**Comment by maxbates** [comment]
forgot password + reset are both updated

**Comment by cojofra** [comment]
The modals now show a scroll bar
screen shot 2017-01-19 at 6 01 23 pm
I think just `overflow-y` needs to be set to `auto` - at least that fixed it in browser.

**Comment by cojofra** [comment]
happy to merge though since it will only go on dev. we also have to update all the screenshots to remove the egf references and I just finished all animations in jquery. let me know if you prefer that in a different branch or add onto this one.

**Comment by maxbates** [comment]
modals need to be able to scroll in the case when they are longer than the window. `overflow: auto` will show the overflow. It is possible to wrap in a container div to hide the scrolling.... when were you seeing them? 

jquery animation can go into this branch, i can help clean it up

**Comment by cojofra** [comment]
tried to reproduce the scrollbar phenotype at home and I can't. When I got it it was on my extended screen with different resolution. I guess will try one last time tomorrow.

File: src/components/authentication/ForgotModal.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    });
+  }
+
+  actions = [{
+    text: 'Send Request',
+    disabled: () => !(this.state.email),
+    onClick: () => this.onForgot(),
+  }];
+
+  render() {
```
Comment (duncanmeech) [inline src/components/authentication/ForgotModal.js]: Needs to block or handle the submit event ( currently it refreshes the page and submit ).

**Comment by maxbates** [comment]
done

**Pull request merged**